classes/class-dintero-checkout-callback.php

```diff
@@ -282,21 +282,22 @@ public static function is_localhost() {
 	 * @return int
 	 */
 	public function get_order_id_from_reference( $merchant_reference ) {
```

Comment:
You have this same method in a couple of different places.

class-dintero-checkout-callback.php - line 284
classes/class-dintero-checkout-confirmation.php - line 153

I suggest moving this to a function in the functions.php file instead, so we ensure the logic is the same in all places.

I would also suggest adding a test at the end to ensure the order we retrieve is actually the correct order, since if the query fails due to bad formatting it will just return the last post of the type you are querying.

For example:
```
// Validate that the order actual has the metadata we're looking for, and that it is the same.
$meta_value = $order->get_meta( '_dintero_merchant_reference', true );

// If the meta value is not the same as the merchant reference, return null.
if ( $meta_value !== $merchant_reference) {
    return null;
}
```

Also why does it return 0 in some cases but null in others?